### PR TITLE
Add workflow to create version labels on new releases

### PR DESCRIPTION
**Pull request by lucasbento**
## Summary

This PR adds the following GitHub actions:

- https://github.com/lucasbento/core-workflow-create-version-label - Create version labels (such as `Version: 0.65.1`) whenever there's a new release;
- https://github.com/lucasbento/core-workflow-apply-version-label - Apply a version label to opened/edited issues based on the version mentioned on the issue body.

## Changelog

N/A.

## Few things to keep in mind

- A label named "Version: unspecified" must be created;
- The GitHub action to create labels will only run when there's a new release or if one is edited, it will then create all the other labels bases on the previous releases until it encounters one that already exists.

---

Example of issue with version label: https://github.com/lucasbento/test-issue-forms/issues/4

### CODE REVIEW

**Comment by cortinico** [comment]
>     * Create version labels (such as `Version: 0.65.1`) whenever there's a new release;

Was this approach discussed somewhere? Is there a reason why we're not using Milestone instead?

**Comment by lucasbento** [comment]
> Was this approach discussed somewhere? Is there a reason why we're not using Milestone instead?

It was discussed on Discord, not sure about the milestones, perhaps @lunaleaps can chime in.

**Comment by lunaleaps** [comment]
Hey @cortinico, yea as @lucasbento mentioned we've been talking about deprecating upgrade-support repo to have upgrade issues reported in this repo. We want to get signal of how many issues come up when people attempt to migrate to a specific version so we can feedback on how stable our releases are.

From looking at milestones, I think the use case seems to be more about tracking a specific goal -- or am I misunderstanding?

File: .github/ISSUE_TEMPLATE/release_blocker_form.yml [hunk 1]
```diff
@@ -23,10 +23,10 @@ body:
     validations:
       required: true
   - type: textarea
-    id: environment
+    id: react-native-info
     attributes:
-      label: Developer Environment
-      description: Please list relevant versions of system, tooling. Ex. OS, processor, Xcode, etc.
+      label: Output of `react-native info`
+      description: Run `react-native info` in your terminal, copy and paste the results here.
```
Comment (lunaleaps) [inline .github/ISSUE_TEMPLATE/release_blocker_form.yml:29]: Oh yes! Great idea

**Comment by cortinico** [comment]
Sorry I didn't wanted to sidetrack your original work. If you already spent time discussing on this, feel free to merge it as it is.

I anyway think that using milestones for this kind of work would be better for a variety of reasons:

> From looking at milestones, I think the use case seems to be more about tracking a specific goal -- or am I misunderstanding?

That's true, if used on PRs. Milestone are basically "version" information that can be attached to an issue or a PR. 

Say for a potential milestone "0.67.0", I would use it in a following way:
* Mark a PR with that milestone if I intend to ship that code in version "0.67.0"
* Mark a issue/feature-request with that milestone if I intend to release that feature in version "0.67.0"
* Mark a bug with that milestone, if want to specify which version introduced that bug, say "0.67.0" <- That's apparently the use case we're looking for.

The problem that I see with the current approach are the following:
* The label space will be polluted with a lot of version-specific labels. We already have 188 labels, and we will end up having way more. Milestones instead can be "closed" and they won't show up in the dropdown selector anymore (but they will still be searchable).
* We're adding two new Github Actions from a 3rd party users (nothing against you @lucasbento), but this is a part of infra that we will have to maintain in the long run. 
* I believe this could have been implemented as part of the `react-native-bot`: https://github.com/hramos/react-native-bot/.

**Comment by lunaleaps** [comment]
@cortinico Thanks for sharing your thoughts! For sure good points and good things to address. At this stage I think labels work well because internal tooling already scrapes this information and I'm not sure if milestones are. I think this is something we can ask about but I already have some tables built out for getting this information.

Is there a cause for concern if there are too many labels? Not exactly sure the cost except for maybe some upper bound of labels that Github allows? Tried searching around but can't find. 

Re: react-native-bot, would that not be an actor that would still need to perform these actions? We can look into conforming the tools -- first want to focus on use-case

**Pull request closed without merging**